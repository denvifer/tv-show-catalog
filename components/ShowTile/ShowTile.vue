<script setup lang="ts">
import type { AppTypes } from "~/types/appTypes";
import { computed } from "vue";

const props = defineProps<{ tile: AppTypes.ShowTile }>();

const attributes = computed(() => [
  {
    icon: "pi-trophy",
    text: props.tile.rating || "-",
  },
  {
    icon: "pi-globe",
    text: props.tile.language || "-",
  },
  {
    icon: "pi-calendar",
    text: props.tile.premiered?.substring(0, 4) || "-",
  },
]);
</script>
<template>
  <nuxt-link :to="`/show/${tile.id}`" class="block no-underline">
    <div
      :class="[
        styles.container,
        'relative shrink-0 flex p-3 gap-3 bg-color-tile-background/[0.8] rounded overflow-hidden',
      ]"
    >
      <div class="flex grow gap-3 overflow-hidden">
        <div
          :class="[
            styles.imageWrapper,
            'relative h-full shrink-0 bg-color-canvas-background rounded overflow-hidden',
          ]"
        >
          <img
            v-if="tile.imageMedium"
            :src="tile.imageMedium"
            :class="styles.image"
            :alt="tile.name"
          />
        </div>
        <section class="min-w-0">
          <div>
            <h3 class="font-bold">
              {{ tile.name }}
            </h3>
          </div>
          <div
            v-for="attribute in attributes"
            :class="['flex', 'gap-2', 'items-center']"
          >
            <i :class="['pi', attribute.icon]"></i>{{ attribute.text }}
          </div>
        </section>
      </div>
    </div>
  </nuxt-link>
</template>
<style module="styles">
.container {
  width: var(--catalog-tile-width);
  aspect-ratio: 2;
}

.imageWrapper {
  aspect-ratio: 8/11;
}

.image {
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  object-fit: cover;
}
</style>
