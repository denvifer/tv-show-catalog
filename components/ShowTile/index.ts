export { default as ShowTile } from "./ShowTile.vue";
