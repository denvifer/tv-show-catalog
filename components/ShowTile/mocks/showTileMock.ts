import type { AppTypes } from "~/types/appTypes";

export const showTileMock: AppTypes.ShowTile = {
  id: 169,
  name: "Breaking Bad",
  rating: 9.2,
  premiered: "2008-01-20",
  language: "English",
  imageMedium:
    "https://static.tvmaze.com/uploads/images/medium_portrait/501/1253519.jpg",
};
