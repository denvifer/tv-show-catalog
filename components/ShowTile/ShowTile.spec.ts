import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";

import ShowTile from "./ShowTile.vue";
import { showTileMock } from "./mocks";

describe("ShowTile", () => {
  it("matches a snapshot", async () => {
    const component = await mount(ShowTile, {
      props: {
        tile: showTileMock,
      },
    });
    expect(component.html()).toMatchSnapshot();
  });
});
