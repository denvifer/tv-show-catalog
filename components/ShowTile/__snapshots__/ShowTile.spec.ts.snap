// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ShowTile > matches a snapshot 1`] = `
"<nuxt-link to="/show/169" class="block no-underline">
  <div class="_container_41e777 relative shrink-0 flex p-3 gap-3 bg-color-tile-background/[0.8] rounded overflow-hidden">
    <div class="flex grow gap-3 overflow-hidden">
      <div class="_imageWrapper_41e777 relative h-full shrink-0 bg-color-canvas-background rounded overflow-hidden"><img src="https://static.tvmaze.com/uploads/images/medium_portrait/501/1253519.jpg" class="_image_41e777" alt="Breaking Bad"></div>
      <section class="min-w-0">
        <div>
          <h3 class="font-bold">Breaking Bad</h3>
        </div>
        <div class="flex gap-2 items-center"><i class="pi pi-trophy"></i>9.2</div>
        <div class="flex gap-2 items-center"><i class="pi pi-globe"></i>English</div>
        <div class="flex gap-2 items-center"><i class="pi pi-calendar"></i>2008</div>
      </section>
    </div>
  </div>
</nuxt-link>"
`;
